### DISCUSSION TITLE
BUILD FAILED (FreeBSD 12.1-RELEASE using ruby-build 20191205)

### DISCUSSION THREAD

**Discussion by tibikuera** [rbenv/ruby-build#1844] [Build failures]
I attempted to install ruby 2.6.5 with ruby-build on a 32-bit powerpc with freebsd and came across the following:

[ruby-build.20191225171610.23660.log](https://github.com/rbenv/ruby-build/files/4000713/ruby-build.20191225171610.23660.log)

```
root@machine:~ # rbenv install 2.6.5
Downloading ruby-2.6.5.tar.bz2...
-> https://cache.ruby-lang.org/pub/ruby/2.6/ruby-2.6.5.tar.bz2
Installing ruby-2.6.5...

BUILD FAILED (FreeBSD 12.1-RELEASE using ruby-build 20191205)

Inspect or clean up the working tree at /tmp/ruby-build.20191225171610.23660.WmQdcU
Results logged to /tmp/ruby-build.20191225171610.23660.log

Last 10 log lines:
gmake[1]: Entering directory '/tmp/ruby-build.20191225171610.23660.WmQdcU/ruby-2.6.5'
gmake[2]: Entering directory '/tmp/ruby-build.20191225171610.23660.WmQdcU/ruby-2.6.5/ext/-test-/arith_seq/extract'
compiling extract.c
linking shared-object -test-/arith_seq/extract.so
../../../../libruby.so: file not recognized: File format not recognized
gmake[2]: *** [Makefile:267: ../../../../.ext/powerpc-freebsd12.1/-test-/arith_seq/extract.so] Error 1
gmake[2]: Leaving directory '/tmp/ruby-build.20191225171610.23660.WmQdcU/ruby-2.6.5/ext/-test-/arith_seq/extract'
gmake[1]: *** [exts.mk:91: ext/-test-/arith_seq/extract/all] Error 2
gmake[1]: Leaving directory '/tmp/ruby-build.20191225171610.23660.WmQdcU/ruby-2.6.5'
gmake: *** [uncommon.mk:286: build-ext] Error 2
root@machine:~ # 
```

**Comment by marvindanig** [comment]
Same issue for me as well.

**Comment by goosnarrggh** [comment]
Repeatable here with ruby-build 20191225 on 32-bit powerpc FreeBSD 12.1 for both 2.6.5 and 2.5.7. Same complaint about libruby.so.

(For what it's worth, I know this is not the correct venue for this extra bit of information, but the FreeBSD 12.1 "port" build of ruby 2.6 for 32-bit powerpc also fails, but in a different way: the bootstrap build of miniruby always segfaults under even the most trivial use cases.)

**Comment by metalefty** [comment]
See also: https://svnweb.freebsd.org/changeset/ports/522532

It looks PowerPC requires GCC to build Ruby.

**Comment by duncan-bayne** [comment]
Repeatable for me on FreeBSD 12.1 AMD64, with Ruby 2.5.1 and Ruby 2.7.1.  My log stops abruptly at the following line, though:

`ln -sf ../../../.ext/include/x86_64-freebsd12.1/rb_mjit_min_header-2.7.1.h include/ruby-2.7.0/x86_64-freebsd12.1/rb_mjit_min_header-2.7.1.h`
